target/linux/ath79/dts/qca9558_sophos_ap.dtsi

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+/*
+ * The hardware of this board family is most likely shared with other devices
+ * from other manufacturers. Edimax appear to be the actual OEM.
+ *
+ * Sophos use the same exact board for the AP55C/AP100C, and AP55/AP100.
+ * Yes, this means your AP55C is a 3x3 AP with a software lock, and your
+ * AP55 is an AP100 with one missing antenna pigtail.
+ *
+ * AP55 and AP55C boards have different physical layouts, but are logically
+ * almost identical. AP55/100 have a USB port which is DNF on the C models.
+ */
+
+#include "qca955x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	aliases {
+		led-boot = &led_status_green;
+		led-failsafe = &led_status_red;
+		led-running = &led_status_green;
+		led-upgrade = &led_status_red;
+		label-mac-device = &eth0;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status_green: status_green {
+			label = "green:status";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		led_status_red: status_red {
+			label = "red:status";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_usb_power: gpio_usb_power {
+			status = "disabled";
+			gpio-export,name = "power:usb";
+			gpio-export,output = <1>;
+			gpios = <&gpio 11 GPIO_ACTIVE_HIGH>;
+		};
+ 	};
```

Comment:
I think the correct approach would be to use a fixed regulator:
```
	reg_usb_vbus: reg_usb_vbus {
		compatible = "regulator-fixed";
		regulator-name = "usb_vbus";
		regulator-min-microvolt = <5000000>;
		regulator-max-microvolt = <5000000>;
		gpio = <&gpio 11 GPIO_ACTIVE_HIGH>;
		enable-active-high;
	};
```

In the USB node you can link the regulator:
```
	vbus-supply = <&reg_usb_vbus>;
```

Reply:
I spent hours trying to work out how to do a GPIO-controlled regulator with a fixed output and somehow _completely_ missed that `regulator-fixed` has an enable GPIO property... Thankyou! Will make this change as well as test on 5.10.

Reply:
Done :) - I had to copy some things from one of the tp-link devices to make power control work (hub port node) - there's an issue in qca955x.dtsi with the PHY names being wrong/noncompliant with mainline (they should just be called `usb`) but that doesn't stop this working & I feel like that should be its own PR.